firefox_desktop/views/urlbar_events.view.lkml

```diff
@@ -1,223 +1,331 @@
-  sql: COUNTIF(${is_terminal} and ${session_action_type} = "engaged");;
-  type: number
-}
+  measure: urlbar_clicks {
+    group_label: "Urlbar Metrics"
+    description: "Count of clicks on any result shown in the urlbar dropdown menu"
+    sql: COUNT(DISTINCT(CASE WHEN ${is_terminal} and ${event_action} = "engaged" THEN ${event_id} ELSE NULL END));;
+    type: number
+  }
 
-measure: urlbar_impressions {
-  group_label: "Urlbar Metrics"
-  description: "The number of times a user exits the urlbar dropdown menu, either by abandoning the urlbar, engaging with a urlbar result, or selecting an annoyance signal that closes the urlbar dropdown menu"
-    sql: COUNTIF(${is_terminal});;
+  measure: urlbar_impressions {
+    group_label: "Urlbar Metrics"
+    description: "The number of times a user exits the urlbar dropdown menu, either by abandoning the urlbar, engaging with a urlbar result, or selecting an annoyance signal that closes the urlbar dropdown menu"
+    sql: COUNT(DISTINCT(CASE WHEN ${is_terminal} THEN ${event_id} ELSE NULL END));;
     type: number
-}
+  }
 
-measure: urlbar_CTR {
-  group_label: "Urlbar Metrics"
-  description: "Clicks / Impressions"
+  measure: urlbar_CTR {
+    group_label: "Urlbar Metrics"
+    description: "Clicks / Impressions"
     sql: safe_divide(${urlbar_clicks}, ${urlbar_impressions});;
     type: number
-}
+  }
 
-measure: urlbar_annoyances {
-  group_label: "Urlbar Metrics"
-  description: "Count of clicks on annoyance signals across all results shown in the urlbar dropdown menu"
-    sql: COUNTIF(${session_action_type} = "annoyance");;
+  measure: urlbar_annoyances {
+    group_label: "Urlbar Metrics"
+    description: "Count of clicks on annoyance signals across all results shown in the urlbar dropdown menu"
+    sql: COUNT(DISTINCT(CASE WHEN ${event_action} = "annoyance" THEN ${event_id} ELSE NULL END));;
     type: number
-}
+  }
 
-  measure: addon_clicks {
-    group_label: "Addon"
-    sql: COUNTIF(product_engaged_result_type = "add_on" and ${is_terminal} and ${session_action_type} = "engaged");;
+  measure: autofill_clicks {
+    group_label: "Autofill"
+    sql: COUNT(DISTINCT(CASE WHEN ${product_selected_result} = "autofill" and ${is_terminal} and ${event_action} = "engaged" THEN ${event_id} ELSE NULL END));;
     type: number
   }
 
-  measure: addon_impressions {
-    group_label: "Addon"
-    description: "The number of times a user exits the urlbar dropdown menu while an addon result was visible"
-    sql: COUNTIF(${num_add_on_impressions} > 0 and ${is_terminal});;
+  measure: autofill_impressions {
+    group_label: "Autofill"
+    description: "The number of times a user exits the urlbar dropdown menu while an Autofill result was visible"
+    sql: COUNT(DISTINCT(CASE WHEN ${is_terminal} and ARRAY_LENGTH(ARRAY( SELECT 1 FROM UNNEST(${results}) AS e WHERE e.product_result_type = "autofill")) > 0 THEN ${event_id} ELSE NULL END));;
```

Comment:
Can we reuse the previous SQL here?

Reply:
If we keep the `count distinct` can we use `${num_add_on_impressions} > 0` instead of the `array_length` statement?

Reply:
done!